src/main/java/co/cask/bigquery/BigQuerySink.java

```diff
@@ -264,12 +307,12 @@ private void decodeSimpleTypes(JsonObject json, String name, Object object, Sche
           try {
             json.addProperty(name, Integer.parseInt(value));
           } catch (NumberFormatException e) {
-            throw new RecordConvertorException(
+            throw new RecordConverterException(
               String.format("Unable to convert '%s' to integer for field name '%s'", value, name)
             );
           }
         } else {
-          throw new RecordConvertorException(
+          throw new RecordConverterException(
```

Comment:
almost all of this method is not needed. It should not be checking the object type after it knows the schema. If the schema says it's an int, the value should be an Integer.

This makes me wonder what sort of input was attached to this sink that the code is expecting invalid records...

Reply:
yea I agree, I will clean it up as part of https://issues.cask.co/browse/CDAP-14187